### ISSUE TITLE
[Bug] Missing support for multipart/related and multipart/mixed in REST API plugin

### ISSUE THREAD

**Issue by nidhi-nair** [appsmithorg/appsmith#7458]
Internal slack discussion: https://theappsmith.slack.com/archives/C01HXJPFWBC/p1631074058090300

**Comment by danieldare** [comment]
@nidhi-nair Please Kindly add a description to this as I'm not able to open the link

**Comment by nidhi-nair** [comment]
Copying context from Slack:

This is coming from [a recent user request](https://discord.com/channels/725602949748752515/883604745418330132/883744629185474561) to use Google Drive API for file uploads. Today we support the mime type multipart/form-data  which is how files and text fields are sent across. What's missing in this is that the multipart universe is actually quite huge. Other mime types include multipart/mixed , multipart/related  and [a bunch of others that I never knew about](https://docs.microsoft.com/en-us/previous-versions/exchange-server/exchange-10/ms527355(v=exchg.10)?redirectedfrom=MSDN).
[Google Drive API documentation](https://developers.google.com/drive/api/v3/manage-uploads) states that their requests need to be sent with the multipart/related  Content-type header. There is no systemic way to choose this in our REST API plugin today. One work around that users can use to achieve this is to use the Raw  option for request body, manually populate the multipart request (which is quite ugly but possible to do because of our lovely dynamic bindings), and fill in file details there as well.
However, this would still fail on the backend because our version of Spring does not yet have the multipart/related  Media type in supported types. We will have to upgrade to [5.2.5](https://github.com/spring-projects/spring-framework/issues/24582) at a minimum for this to be available. We might have to potentially work with the folks at Spring for every new type we discover a use case for. That would still restrict us to this one new type though which could be a future problem to solve. TBH, the rest of the options on that link aren't things I've seen a lot so I'd say we can put this one off.
Postman btw, [also hasn't created a systemic solve for non-form-data sub types](https://github.com/postmanlabs/postman-app-support/issues/8817), their work around is similar to ours with the raw body.
TLDR; what needs changing:
Ability on the UI to pick a subtype in multipart
2a.  Upgrade Spring framework dependency to support one of these types.
           OR
2b.  Figure out a way to not have WebClient be dependent on the pre-defined types (down-vote from my end for this)
